vortex-array/src/array/compute.rs

```diff
@@ -0,0 +1,12 @@
+use crate::compute::{ComputeFn, KernelRef};
+
+/// A trait used to register static kernels for known compute functions.
+/// Dynamic kernels must be returned via the `_find_kernel` method.
+pub trait ArrayComputeImpl {
+    const FILTER: Option<KernelRef> = None;
+
+    /// Fallback implementation to lookup compute kernels at runtime.
+    fn _find_kernel(&self, _compute_fn: &dyn ComputeFn) -> Option<KernelRef> {
```

Comment:
don't you want to pass arguments here as well?

Reply:
Eh, not sure. Decided against because all kernels return `Option<Output>`. We may want to add later if a use-case comes up